packages/core/test/unit/patchers/aws_p.test.js

```diff
@@ -252,9 +252,93 @@ describe('AWS patcher', function() {
 
       awsRequest.emitter.emit('complete');
 
-      setTimeout(function() {
+      setTimeout(function () {
         getCauseStub.should.have.been.calledWithExactly(awsRequest.response.httpResponse.statusCode);
-        closeStub.should.have.been.calledWithExactly(sinon.match({ message: error.message, name: error.code}), true);
+        closeStub.should.have.been.calledWithExactly(sinon.match({ message: error.message, name: error.code }), true);
+        done();
+      }, 50);
+    });
+  });
+
+  describe('#captureAWSRequest-Unsampled', function () {
+    let awsClient, awsRequest, MyEmitter, sandbox, segment, stubResolve, stubResolveManual, sub;
+
+    before(function () {
+      MyEmitter = function () {
+        EventEmitter.call(this)
+      }
+
+      awsClient = {
+        customizeRequests: function customizeRequests (captureAWSRequest) {
+          this.call = captureAWSRequest;
+        },
+        throttledError: function throttledError () {}
+      }
+      awsClient = awsPatcher.captureAWSClient(awsClient);
+
+      util.inherits(MyEmitter, EventEmitter);
+    })
+
+    beforeEach(function () {
+      sandbox = sinon.createSandbox();
+
+      awsRequest = {
+        httpRequest: {
+          method: 'GET',
+          url: '/',
+          connection: {
+            remoteAddress: 'localhost'
+          },
+          headers: {}
+        },
+        response: {}
+      };
+
+      awsRequest.on = function (event, fcn) {
+        if (event === 'complete') {
+          this.emitter.on(event, fcn.bind(this, this.response));
+        } else {
+          this.emitter.on(event, fcn.bind(this, this));
+        }
+        return this;
+      };
+
+      awsRequest.emitter = new MyEmitter();
+
+      segment = new Segment('testSegment', traceId);
+      sub = segment.addNewSubsegmentWithoutSampling('subseg');
+
+      stubResolveManual = sandbox.stub(contextUtils, 'resolveManualSegmentParams');
+      stubResolve = sandbox.stub(contextUtils, 'resolveSegment').returns(segment);
+      sandbox.stub(segment, 'addNewSubsegmentWithoutSampling').returns(sub);
+    });
+
+    afterEach(function () {
+      sandbox.restore();
+    });
+
+    it('should log an info statement and exit if parent is not found on the context or on the call params', function (done) {
+      stubResolve.returns();
+      const logStub = sandbox.stub(logger, 'info');
+
+      awsClient.call(awsRequest);
+
+      setTimeout(function () {
+        logStub.should.have.been.calledOnce;
```

Comment:
If it is not supposed to sample, then should this have emitted?

Reply:
I believe this is unrelated - just checking if a log is produced. There is another test right below checking the trace header being sent out